mam/v2/sponge/spongos_types.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * MAM is based on an original implementation & specification by apmi.bsu.by
+ * [ITSec Lab]
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __MAM_V2_SPONGE_SPONGOS_TYPES_H__
+#define __MAM_V2_SPONGE_SPONGOS_TYPES_H__
+
+#include "mam/v2/defs.h"
+#include "mam/v2/sponge/sponge.h"
+#include "mam/v2/trits/trits.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct mam_spongos_s {
+  mam_sponge_t *sponge;
```

Comment:
Investigations:
- Does this need to be serialized ?
- Is this shared ?
- Static / pointer ?

Reply:
We need to serialize it separately, not as part as WOTS so it seems